addons/website_hr_recruitment/static/src/js/website_hr_applicant_form.js

```diff
@@ -25,6 +27,62 @@ publicWidget.registry.hrRecruitment = publicWidget.Widget.extend({
         }
     },
 
+    async _onFocusOutMail (ev) {
+        const email = $(ev.currentTarget).val();
+        if (!email) {
+            $(ev.currentTarget).removeClass('border-warning');
+            $('#email-message').removeClass('alert-warning').hide();
+            return;
+        }
+        const job_id = $('#recruitment7').val();
+        const data = await rpc('/website_hr_recruitment/check_recent_application',
+            {
+                email: email,
+                job_id: job_id,
+            });
+        if (data.applied_same_job)  {
+            $('#email-message').removeClass('alert-warning').hide();
+            $(ev.currentTarget).addClass('border-warning');
+            $('#email-message').text(_t(data.message)).addClass('alert-warning').show();
+        } else if (data.applied_other_job)  {
+            $('#email-message').removeClass('alert-warning').hide();
+            $(ev.currentTarget).addClass('border-warning');
+            $('#email-message').text(_t("You already applied to another position recently. You can continue if it's not a mistake.")).addClass('alert-warning').show();
+        } else {
+            $(ev.currentTarget).removeClass('border-warning');
+            $('#email-message').removeClass('alert-warning').hide();
+        }
```

Comment:
This code is very redundant across the different methods. You should probably create a method `checkRedundant` that takes the target of the field as an argument.